### PR TITLE
Add support for `filters` to `Vector#delete`

### PR DESCRIPTION
**Pull request by pcreux**
Add support for `filters` to `Vector#delete`.

I added a ruby script to test it because the test suite requires access to a specific index that I don't have access to.

Feel free to close this PR as the code change is minimal.

### CODE REVIEW

**Comment by pcreux** [comment]
@ScotterC Done! Thank you.

**Comment by pcreux** [comment]
> For next time, you can spin up the test indexes and shut them down using ruby spec/support/setup.rb start and stop

I believe that the index name is still hard-coded in the test suite though. 🤔

**Comment by ScotterC** [comment]
Oh yea. The VCR cassettes do. Maybe if I have VCR filter out the index name it'll run agnostically to the index name

**Comment by pcreux** [comment]
That should work yes. 🙂

File: spec/pinecone/vector_spec.rb [hunk 1]
```diff
@@ -50,6 +50,18 @@
         expect(response.code).to eq(200)
       end
     end
+
+    describe "successful response with filters" do
+      let(:response) {
+        index.delete(filter: { "genre": "comedy"})
```
Comment (pcreux) [inline spec/pinecone/vector_spec.rb:56]:
```suggestion
        index.delete(filter: {genre: "comedy"})
```

**Comment by pcreux** [comment]
@ScotterC The PR is ready to go. You'd just need to run it on your machine to record the cassette.

**Comment by ScotterC** [comment]
Thanks for the contribution!

**Pull request merged**